lib/ain-evm/src/storage/cache.rs

```diff
@@ -3,30 +3,35 @@ use std::{borrow::ToOwned, num::NonZeroUsize, sync::RwLock};
 use ethereum::{BlockAny, TransactionV2};
 use ethereum_types::{H256, U256};
 use lru::LruCache;
+use parking_lot::Mutex;
 
 use super::traits::{BlockStorage, Rollback, TransactionStorage};
 use crate::Result;
 
 #[derive(Debug)]
 pub struct Cache {
-    transactions: RwLock<LruCache<H256, TransactionV2>>,
-    blocks: RwLock<LruCache<U256, BlockAny>>,
-    block_hashes: RwLock<LruCache<H256, U256>>,
+    transactions: Mutex<LruCache<H256, TransactionV2>>,
+    blocks: Mutex<LruCache<U256, BlockAny>>,
+    block_hashes: Mutex<LruCache<H256, U256>>,
     latest_block: RwLock<Option<BlockAny>>,
+    codes: Mutex<LruCache<H256, Vec<u8>>>,
```

Comment:
I think this would be more descriptive.
```suggestion
    contract_code: Mutex<LruCache<H256, Vec<u8>>>,
```